test/src/e2e_vm_tests/test_programs/should_fail/where_clause_impls/test.toml

```diff
@@ -1,11 +1,10 @@
 category = "fail"
 
+# check: $()x: a.x.my_add(b.x),
+
 # check: $()x: a.x.my_add(b.x),
 # nextln: $()No method named "my_add" found for type "T".
 # check: $()y: a.y.my_add(b.y),
 
 # check: $()y: a.y.my_add(b.y),
 # nextln: $()No method named "my_add" found for type "T".
-
-# check: $()let baz = foo.my_add(bar);
-# nextln: $()No method named "my_add" found for type "MyPoint<u32>".
```

Comment:
There are some slight changes to control flow because refactoring things makes some functions infallible, allows us to detect cases where errors always/never happen.

In this case we're bailing because of the previous errors, which I think is reasonable given how the test code is broken.

If you repair the impl in the test so that the previous two errors are fixed, this error will show up, which I think is decent enough.

Reply:
I think this change is sufficiently small not to pay too much attention to it, given the size of this refactoring :)